doc/holochain_101/src/zome/entry_type_definitions.md

```diff
@@ -208,97 +152,27 @@ Further reading can be found [here](./entry_validation.md).
 
 ### Putting It All Together
 
-Taken all together, use of the `entry!` macro may look something like the following:
-
-```rust
-...
-entry!(
-    name: "post",
-    description: "A blog post entry which has an author",
-    sharing: Sharing::Public,
-    validation_package: || {
-        ValidationPackageDefinition::Entry
-    },
-    validation: |validation_data: EntryValidationData<Post>| {
-        Ok(())
-    }
-)
-```
-
-This can be embedded directly inside of the entries array of the `define_zome!`, like so:
-```rust
-...
-define_zome! {
-    entries: [
-        entry!(
-            name: "post",
-            description: "A blog post entry which has an author",
-            sharing: Sharing::Public,
-
-            validation_package: || {
-                ValidationPackageDefinition::Entry
-            },
-            validation: | _validation_data: EntryValidationData<Post>| {
-                Ok(())
-            }
-        )
-    ]
-
-    init: || {
-        Ok(())
-    }
-
-    validate_agent: |validation_data : EntryValidationData::<AgentId>| {
-        Ok(())
-    }
-
-    functions: []
-
-    capabilitites: {}
-}
-```
+Taken all together, defining an entry inside the `#[zome]` module using the `entry!` macro may look something like the following:
 
-If there is only entry type, this can be fine, but if there are multiple, this can hurt readability of the source code. You can wrap the entry type definition in a function, and call it in `define_zome!`, like so:
 ```rust
 ...
-fn post_definition() -> ValidatingEntryType {
+#[entry_def]
+fn post_entry_def() -> ValidatingEntryType {
     entry!(
         name: "post",
         description: "A blog post entry which has an author",
         sharing: Sharing::Public,
-
         validation_package: || {
-            hdk::ValidationPackageDefinition::Entry
+            ValidationPackageDefinition::Entry
         },
-
-        validation: |_validation_data: Validation<Post>| {
+        validation: |validation_data: EntryValidationData<Post>| {
```

Comment:
double-checking: is rust okay with not using the param validation_data? Would this benefit from a preceeding underscore (i.e. `_validation_data`)?